mesonbuild/dependencies/dev.py

```diff
@@ -700,6 +701,95 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
 packages['jdk'] = JDKSystemDependency
 
 
+class DiaSDKSystemDependency(SystemDependency):
+
+    def _try_path(self, diadir: str, cpu: str) -> bool:
+        if not os.path.isdir(diadir):
+            return False
+
+        include = os.path.join(diadir, 'include')
+        if not os.path.isdir(include):
+            mlog.error('DIA SDK is missing include directory:', include)
+            return False
+
+        lib = os.path.join(diadir, 'lib', cpu, 'diaguids.lib')
+        if not os.path.exists(lib):
+            mlog.error('DIA SDK is missing library:', lib)
+            return False
+
+        bindir = os.path.join(diadir, 'bin', cpu)
+        if not os.path.exists(bindir):
+            mlog.error(f'Directory {bindir} not found')
+            return False
+
+        found = glob.glob(os.path.join(bindir, 'msdia*.dll'))
+        if not found:
+            mlog.error("Can't find msdia*.dll in " + bindir)
+            return False
+        if len(found) > 1:
+            mlog.error('Multiple msdia*.dll files found in ' + bindir)
+            return False
+        self.dll = found[0]
+
+        # What to do about version?
+        # At runtime, the DIA interface version used is controlled by GUID passed to CoCreateInstance.
+        # Newer DLLs contain backwards support for all older GUID.
+        # DIA SDK include files only provide GUIDS for the latest version.
+        # Finding out version of current SDK is tricky. The only place where the actual SDK
+        # version is stored, seems to be the Version field in msdia*.dll resources.
+        # To get the just the major version, we could read it from msdiaNNN.dll file name:
+        self.version = 'unknown'
+
+        self.compile_args.append('-I' + include)
+        self.link_args.append(lib)
+        self.is_found = True
+        return True
+
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]) -> None:
+        super().__init__('diasdk', environment, kwargs)
+        self.is_found = False
+
+        compilers = environment.coredata.compilers.host
+        if 'cpp' in compilers:
+            self.compiler = compilers['cpp']
+        elif 'c' in compilers:
+            self.compiler = compilers['c']
+        else:
+            mlog.error('DIA SDK is only supported in C and C++ projects')
+            return
+
+        if self.compiler.id not in ['msvc', 'clang']:
+            mlog.error('DIA SDK is only supported with MSVC compiler')
+            return
+
+        cpu_translate = {'arm': 'arm', 'aarch64': 'arm64', 'x86': '.', 'x86_64': 'amd64'}
+        cpu_family = environment.machines.host.cpu_family
+        if cpu_family not in cpu_translate:
+            mlog.error(f'DIA SDK is not supported for "{cpu_family}" architecture')
+            return
+        cpu = cpu_translate[cpu_family]
+
+        diadir = os.path.join(os.environ.get('VSInstallDir'), 'DIA SDK')
```

Comment:
This will fail if `os.environ.get()` returns `None`. I think I would handle this by:
```python
try:
    diadir = os.path.join(os.environ['VSInstallDir'], 'DIA SDK')
except KeyError:
    mlog.log('VSInstallDir is not set in environment')
    return
```